### DISCUSSION TITLE
Adjust the support of Spark 2.x version

### DISCUSSION THREAD

**Discussion by baiyangtx** [apache/amoro#1325] [General]
# Backgroud
The community is pushing forward the upgrading  the iceberg version to 1.1.0,  #989  
And in iceberg 1.1.0,  iceberg changes the spark2 module to spark 2.4, but Spark2.x version supported by Arctic is 2.3, and Arctic dose some extra things to support Spark SQL in spark 2.x verison. 

I have bean researched that the spark 2.x version and features supported by datalake formats.

| Format | Spark2.x Version | DataFrameAPI | Spark SQL |
|---------|-------------------|-----------------|-------------|
| Iceberg |  2.4                |  √                 |      X      |
| Hudi |  2.4.3+                |  √                 |      √      |
| Paimon|  2.4                |  √                 |      X      |
| Delta |  NotSupported in lasted version | | |

There several options:

1. Remove support of Spark 2.x, Arctic will only support Spark 3.1+ 
2. Change the Spark 2.3 support to 2.4, and support Dataframe API only.
3. Change the Spark 2.3 support to 2.4, and support Dataframe API and Spark SQL.

**Comment by dupen01** [comment]
It doesn't matter to me because I use spark3😁

**Comment by GavinH1984** [comment]
Vote for option 1~~~😁

**Comment by pan3793** [comment]
Vote for option 1. Spark 2 was EOL for quite a long time

**Comment by Aireed** [comment]
Vote for option 1

**Comment by zhongqishang** [comment]
Vote for option 1

**Comment by zhoujinsong** [comment]
We have stopped supporting spark 2.x for Mixed formats.
